### PR TITLE
docs: enable code coverage.

### PR DESCRIPTION
**Pull request by datamweb**
Explain about the activation `coverage ` that is required to run the **unit test**.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ If running under OS X or Linux, you can create a symbolic link to make running t
 	> ln -s ./vendor/bin/phpunit ./phpunit
 
 You also need to install [XDebug](https://xdebug.org/docs/install) in order
-for code coverage to be calculated successfully.
+for code coverage to be calculated successfully. After installation `XDebug`. You must add ``xdebug.mode=coverage`` in file **php.ini**. Because code coverage needs to be enable.Then do not forget to restart **Apache**.
```
Comment (kenjis) [inline tests/README.md:22]: I believe Apache has nothing to do with. `phpunit` runs on CLI.

Comment (datamweb) [reply tests/README.md:22]: Do you use **XAMPP** or **MAMP** or **WampServer**? Most likely your answer is **no**.
That's why you raised this issue.
I use **XAMPP**. Therefore, after each change in file **php.ini**, **Apache** must be restarted.
I was involved in this for several hours. Finally decide to merge with you.

Comment (kenjis) [reply tests/README.md:22]: I don't use Windows.

Comment (kenjis) [reply tests/README.md:22]: In my understanding, PHP for Web (Apache) and PHP for CLI are different things.
If you run PHP via web, you need to restart Apache if you change the php.ini.
But PHP for CLI reads php.ini every time, so you don't need to do something.

> The configuration file (php.ini) is read when PHP starts up. For the server module versions of PHP, this happens only once when the web server is started. For the CGI and CLI versions, it happens on every invocation. 
https://www.php.net/manual/en/configuration.file.php

I don't see why XAMPP should restart Apache even with the CLI.

Comment (paulbalandan) [reply tests/README.md:22]: > I believe Apache has nothing to do with. `phpunit` runs on CLI.

Agree with this. You do not run tests on the web but on the CLI.

File: tests/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ If running under OS X or Linux, you can create a symbolic link to make running t
 	> ln -s ./vendor/bin/phpunit ./phpunit
 
 You also need to install [XDebug](https://xdebug.org/docs/install) in order
-for code coverage to be calculated successfully.
+for code coverage to be calculated successfully. After installation `XDebug`. You must add ``xdebug.mode=coverage`` in file **php.ini**. Because code coverage needs to be enable.
```
Comment (paulbalandan) [inline tests/README.md:22]:
```suggestion
for code coverage to be calculated successfully. After installing `XDebug`, you must add `xdebug.mode=coverage` in the **php.ini** file to enable code coverage.
```

**Comment by paulbalandan** [comment]
We can just squash on merge.

**Comment by datamweb** [comment]
> @datamweb Thank you for updating.
> 
> In this PR, you have two commits, but one commit is enough for this change. Why don't you squash? See https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

@kenjis Thanks for the link. I think I did it.

File: user_guide_src/source/models/model.rst [hunk 3]
```diff
@@ -295,6 +295,8 @@ the array's values are the values to save for that key:
 
 .. literalinclude:: model/015.php
 
+You can retrieve the last inserted row's primary key using the ``getInsertID()`` method.
+
```
Comment (kenjis) [inline user_guide_src/source/models/model.rst:299]: > @kenjis Thanks for the link. I think I did it.

Yes, you did it!

But this file change has nothing to do with this PR.
You squashed unrelated previous? commit?

Comment (datamweb) [reply user_guide_src/source/models/model.rst:299]: **!!!**,I do not know why this change was applied.
Do I close this PR and open a new PR?

**Comment by kenjis** [comment]
Try this:
```
$ git reset --hard HEAD^  # delete the last commit completely
$ git reset HEAD^  # cancel the last commit
$ git restore user_guide_src/source/models/model.rst  # remove the change of the file
$ git add tests/README.md
$ git commit
```

**Comment by datamweb** [comment]
> 



> Try this:
> 
> ```
> $ git reset --hard HEAD^  # delete the last commit completely
> $ git reset HEAD^  # cancel the last commit
> $ git restore user_guide_src/source/models/model.rst  # remove the change of the file
> $ git add tests/README.md
> $ git commit
> ```

`$ git commit
[CodeCoverageEnable 7ec2bb603] Update from git
 1 file changed, 1 insertion(+), 1 deletion(-)
`
**but** 
```
error: You have not concluded your merge (MERGE_HEAD exists).
hint: Please, commit your changes before merging.
fatal: Exiting because of unfinished merge.
```

**Comment by kenjis** [comment]
@datamweb Did you use `git pull`?

**Comment by datamweb** [comment]
> 
I am using the desktop version of Github. I think that's why.

Screenshot 2022-04-30 023028

**Comment by kenjis** [comment]
I recommend you never use `git pull`.
It creates a lot of troubles.
See https://stackoverflow.com/questions/15316601/in-what-cases-could-git-pull-be-harmful

**Comment by datamweb** [comment]
> 

I'm really tired. I am suffocating in a world of new information. Thanks for link.

**Comment by kenjis** [comment]
Unfortunately, some people are unable to complete Pull Requests because they cannot use git properly.

But don't worry, git is not an easy tool, and it usually takes time to learn.

**Comment by datamweb** [comment]
@kenjis Please check, did I do it right or not?
With your help, I will either find a way or I will build a way.

**Review by kenjis — APPROVED** [review]
LGTM!

**Comment by kenjis** [comment]
@datamweb You did it! Perfect! 👍

**Pull request merged**